Remotion/Security/Core/Metadata/MetadataCache.cs

```diff
@@ -102,10 +102,11 @@ private PropertyInfo NormalizeProperty (PropertyInfo property)
       if (property.DeclaringType == property.ReflectedType)
         return property;
       else
-        return property.DeclaringType.GetProperty (property.Name);
+        // TODO: Add debug notnull assertion
```

Comment:
notnull-assertion.  (not debugnotnull, since the property info could theoretically be passed from uncontrolled external code.